libvips/colour/colour.c

```diff
@@ -245,40 +242,43 @@ vips_colour_build(VipsObject *object)
 
 	if (vips_image_generate(out,
 			vips_start_many, vips_colour_gen, vips_stop_many,
-			in, colour)) {
-		g_object_unref(out);
+			in, colour))
 		return -1;
-	}
 
 	/* Reattach higher bands, if necessary. If we have more than one input
 	 * image, just use the first extra bands.
 	 */
-	for (i = 0; i < colour->n; i++)
+	for (int i = 0; i < colour->n; i++)
 		if (extra_bands[i]) {
-			VipsImage *t1, *t2;
+			VipsImage **t = (VipsImage **) vips_object_local_array(object, 5);
+
+			double max_alpha_before =
+				vips_interpretation_max_alpha(extra_bands[i]->Type);
+			double max_alpha_after =
+				vips_interpretation_max_alpha(out->Type);
+
+			VipsImage *in;
```

Comment:
nit: `in` was already declared outside the loop, perhaps we should use a different variable name here? This would also appease any `-Wshadow` warnings in this file.
```
../libvips/colour/colour.c: In function ‘vips_colour_build’:
../libvips/colour/colour.c:260:36: warning: declaration of ‘in’ shadows a previous local [-Wshadow]
  260 |                         VipsImage *in;
      |                                    ^~
../libvips/colour/colour.c:164:21: note: shadowed declaration is here
  164 |         VipsImage **in;
      |             
```

Reply:
Nice catch, I'll rename it `alpha`.